src/Streamly/Internal/Data/Stream/StreamD.hs

```diff
@@ -2951,8 +2951,10 @@ gbracketIO bef exc aft fexc fnormal =
     -- weak pointer to us.
     {-# INLINE_LATE step #-}
     step _ GBracketIOInit = do
-        r <- bef
-        ref <- newFinalizedIORef (aft r)
+        (r, ref) <- liftBaseOp_ mask_ $ do
+            r <- bef
+            ref <- newFinalizedIORef (aft r)
```

Comment:
It would be nice if you could write up small developer documentation regarding the change or paste a few references right above the code.

Reply:
Thanks for looking into this. I just pushed a few comments.